### PR TITLE
Fix e2e tests settings

### PR DESCRIPTION
**Pull request by flipsimon**
e2e tests are working on ubuntu now at least

### CODE REVIEW

File: test/testcafe/messagelist_tests.ts [hunk 1]
```diff
@@ -187,7 +189,9 @@ function endCode(msg) {
 function MSGinViewportSelector(timeout = 2000): Selector {
   return Selector(
     () => {
-      let raw_labels = document.querySelectorAll('#message-list .msg-body')
+      let raw_labels = document.querySelectorAll(
+        '#message-list .msg-body, .join-button'
+      )
```
Comment (Simon-Laux) [inline test/testcafe/messagelist_tests.ts:194]: we probably need something more advanced here as this can currently only detect video messages and does not necessarily select the last/most recent video message.

also does it work as expected? means that it only passes if the element is visible to the user without scrolling.

Comment (Simon-Laux) [reply test/testcafe/messagelist_tests.ts:194]: https://testcafe.io/documentation/402829/guides/basic-guides/select-page-elements could help here, its the documentation about testcafe selectors

File: scss/main_screen/_navbar_wrapper.scss [hunk 2]
```diff
@@ -83,7 +83,7 @@
     font-size: 18px;
   }
 
-  #main-menu-button {
+  #three-dot-menu-button {
```
Comment (Jikstra) [inline scss/main_screen/_navbar_wrapper.scss:86]: can we leave it as three-dot menu? Because it's not really a main menu ^^

File: src/renderer/components/attachment/menuAttachment.tsx [hunk 3]
```diff
@@ -127,7 +127,7 @@ const MenuAttachment = ({
         }
         position={Position.TOP_LEFT}
       >
-        <Button minimal icon='paperclip' />
+        <Button id='test-attachment-menu' minimal icon='paperclip' />
```
Comment (Jikstra) [inline src/renderer/components/attachment/menuAttachment.tsx:130]: ?

File: src/renderer/components/screens/MainScreen.tsx [hunk 4]
```diff
@@ -240,54 +241,55 @@ export default function MainScreen() {
               </NavbarHeading>
             )}
             {selectedChat.chat && (
-              <span className='views'>
```
Comment (Jikstra) [inline src/renderer/components/screens/MainScreen.tsx:243]: Why are these changes needed?

File: test/testcafe/messagelist_tests.ts [hunk 5]
```diff
@@ -146,10 +149,31 @@ const clickChatByName = async (t, name) => {
...
+  await clickSideBarItem(t, await translate('menu_settings'))
+  await t.expect(Selector('.SettingsIconButton').withText(label).exists).ok()
+  await t.click(Selector('.SettingsIconButton').withText(label))
+}
+
+export async function sendVideoChatInvitation(t) {
+  await t.click('#test-attachment-menu')
+  await t.click(
+    Selector('a.bp3-menu-item').withText(await translate('videochat'))
+  )
```
Comment (Jikstra) [inline test/testcafe/messagelist_tests.ts:176]: You can put functions that you are reusing to helpers.js :)

**Review by Jikstra — COMMENTED** [review]
Some queeestioooons

File: scss/main_screen/_navbar_wrapper.scss (hunk 2 above)
Comment (Simon-Laux) [reply scss/main_screen/_navbar_wrapper.scss:86]: ?

File: src/renderer/components/attachment/menuAttachment.tsx (hunk 3 above)
Comment (Simon-Laux) [reply src/renderer/components/attachment/menuAttachment.tsx:130]: without an id the test script can not easily find it

File: src/renderer/components/screens/MainScreen.tsx (hunk 4 above)
Comment (Simon-Laux) [reply src/renderer/components/screens/MainScreen.tsx:243]: to hide the chat context menu when no chat is selected, basically to fix a bug

File: test/testcafe/messagelist_tests.ts (hunk 5 above)
Comment (Simon-Laux) [reply test/testcafe/messagelist_tests.ts:176]: maybe would make some sense to convert all tests to TS in that case?

**Comment by Simon-Laux** [comment]
there are some message list tests regarding video chat that should fail, because there is a bug in desktop and the test should detect that if it is working correctly

File: src/renderer/components/attachment/menuAttachment.tsx (hunk 3 above)
Comment (Jikstra) [reply src/renderer/components/attachment/menuAttachment.tsx:130]:
```suggestion
        <Button id='attachment-menu' minimal icon='paperclip' />
```

File: test/testcafe/messagelist_tests.ts (hunk 5 above)
Comment (Jikstra) [reply test/testcafe/messagelist_tests.ts:176]: > there are some message list tests regarding video chat that should fail, because there is a bug in desktop and the test should detect that if it is working correctly

Lets keep the needed changes for this pr simple so we can merge soon?

**Comment by Jikstra** [comment]
> there are some message list tests regarding video chat that should fail, because there is a bug in desktop and the test should detect that if it is working correctly

Lets keep the needed changes for this pr simple so we can merge soon?

File: src/renderer/components/attachment/menuAttachment.tsx (hunk 3 above)
Comment (Simon-Laux) [reply src/renderer/components/attachment/menuAttachment.tsx:130]: the idea of the `test-` prefix was to mark it as only used in tests, some other id's are used also by desktop

**Comment by Simon-Laux** [comment]
> Lets keep the needed changes for this pr simple so we can merge soon?

what do the tests matter if they are green, but broken? the goal of this pr was/is to fix the tests

File: test/testcafe/messagelist_tests.ts (hunk 1 above)
Comment (flipsimon) [reply test/testcafe/messagelist_tests.ts:194]: As it is only for the test, we can decide that we won't have multiple videochat invitations in the chat. So in my opinion, it doesn't really matter if the selector theoretically selects all video chat invitations. 
For me it worked as expected. When the videochat invitation was not in sight for the user, the test failed, when I stopped and scrolled down, it passed.

**Comment by flipsimon** [comment]
Added messagelist tests to the workflow. Now, on ubuntu, the e2e tests work as expected: all pass except for the videochat invitation scroll down test.
